examples/BCDR/BackupFromStorageQueue.cs

```diff
@@ -0,0 +1,172 @@
+
+            // Read all settings from Primary Store and update secondary store
+            foreach (ConfigurationSetting setting in primaryAppConfigClient.GetConfigurationSettings(new SettingSelector()))
+            {
+                KeyLabel primaryStoreKeyLabel = new KeyLabel(setting.Key, setting.Label);
+                if(updatedKeyLabels.Contains(primaryStoreKeyLabel))
+                {
+                    // current setting retrieved from primary store needs to be updated in secondary store
+                    secondaryAppConfigClient.SetConfigurationSetting(setting);
+                    log.LogInformation($"Successfully updated key: {setting.Key} label: {setting.Label}");
+                    updatedKeyLabels.Remove(primaryStoreKeyLabel);
+                }
+            }
+
+            // Delete key labels still present in updatedKeyLabels from secondary store
+            foreach (var keyLabel in updatedKeyLabels)
+            {
+                secondaryAppConfigClient.DeleteConfigurationSetting(keyLabel.Key, keyLabel.Label);
+                log.LogInformation($"Successfully deleted key: {keyLabel.Key} label: {keyLabel.Label} from secondary store.");
+            }
+        }
+
+        [FunctionName("BackupFromStorageQueue")]
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            try
+            {
+                log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+                
+                // Create storage queue client
+                QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+
+                // Peek to see if there are events in the queue
+                while (queueClient.PeekMessages().Value.Length > 0 )
+                {
+                    Response<QueueMessage[]> messages = queueClient.ReceiveMessages(maxMessages: 32);
+
+                    // extract list of key+labels from event data
+                    HashSet<KeyLabel> updatedKeyLabels = ExtractKeyLabelsFromEvents(messages.Value, log);
+
+                    // If there are any valid App Configuration events, update secondary store
+                    if(updatedKeyLabels.Count > 0)
+                    {
+                        BackupKeyValues(updatedKeyLabels, log);
+                    }
+
+                    // Delete this batch of events from storage queue
+                    foreach (QueueMessage message in messages.Value)
+                    {
+                        log.LogInformation($"Deleting message: {message.MessageId}");
+                        queueClient.DeleteMessage(message.MessageId, message.PopReceipt);
+                    }
+                }
+            }
+            catch (RequestFailedException exception)
```

Comment:
Should these error handlings be part of `BackupKeyValues`?

Reply:
All right. We have to handle the case when the backup failed, we don't delete the messages.

OK, in case the primary is down, the backup will fail. When this happens, we should just abort and the new attempt will be made when the Azure Function is triggered again next time. We can change `BackupKeyValues` to return bool and decide whether we should abort.

Reply:
Moved this exception handling to BackupKeyValues, which now returns a boolean to indicate if anything failed. If there was any exception thrown by AppConfig, we log the exception and abort the function. Queue messages will be available for processing again after default timeout.